### PR TITLE
Update ticktick extension

### PR DESCRIPTION
**Pull request by Alireza29675**
## Description

When creating a new task, the extension will guess which project the task belongs to. For non-pro users it just acts as before.

## Screencast

screencapture

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Seems to work pretty good @Alireza29675, well done! 🔥 

@viduycheung do you mind checking this?

File: extensions/ticktick/src/create.tsx [hunk 1]
```diff
@@ -74,6 +77,16 @@ export default function TickTickCreate() {
 
   const onListChange: NonNullable<Form.Dropdown.Props["onChange"]> = useCallback((newValue) => {
     LocalStorage.setItem("defaultAddList", newValue);
+    setProjectId(newValue);
+  }, []);
+
+  const onTitleBlur: NonNullable<Form.TextField.Props["onBlur"]> = useCallback(async (newValue) => {
+    const title = newValue.target.value;
+    if (title && environment.canAccess(AI)) {
+      const toast = await showToast(Toast.Style.Animated, "🧠 Guessing project...");
```
Comment (viduycheung) [inline extensions/ticktick/src/create.tsx:86]: debounce guessing project when on change and cancel debounce on blur and guessing project immediately will be better

File: extensions/ticktick/src/create.tsx [hunk 2]
```diff
@@ -89,14 +102,14 @@ export default function TickTickCreate() {
       }
       isLoading={isLoading}
     >
-      <Form.TextField ref={titleRef} id="title" autoFocus title="Title" placeholder="No Title" />
-      <Form.Dropdown ref={listPickerRef} id="list" title="List" onChange={onListChange} defaultValue={defaultAddList}>
+      <Form.TextField ref={titleRef} id="title" onBlur={onTitleBlur} autoFocus title="Title" placeholder="No Title" />
+      <Form.TextArea ref={descRef} id="desc" title="Description" placeholder="" />
+      <Form.DatePicker ref={dueDatePickerRef} id="dueDate" title="Due Date" type={Form.DatePicker.Type.DateTime} />
```
Comment (viduycheung) [inline extensions/ticktick/src/create.tsx:107]: restore the original order

**Review by viduycheung — CHANGES_REQUESTED** [review]
It's a great idea, but there are a few details to improve.
I think this feature should be disabled by default and can manually open by the user.

File: extensions/ticktick/src/create.tsx (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/ticktick/src/create.tsx:86]: That would make sense, I like the idea 🙂

Comment (Alireza29675) [reply extensions/ticktick/src/create.tsx]: Great idea! I put a 500ms debounce. We might hit the rate limit if we type slowly and add multiple tasks in a row, but in normal usage it works fine to me. 🙌

File: extensions/ticktick/src/create.tsx (hunk 2 above)
Comment (Alireza29675) [reply extensions/ticktick/src/create.tsx]: My main motivation for that was that if it's being done by autocomplete, then it probably shouldn't be in the next tab-index. What do you think?

(I've put the original order back in place, so we can keep it as it was or change it we decided to)

**Comment by Alireza29675** [comment]
@pernielsentikaer Settings have been added and are disabled by default 🙌

Should we enable it by default or wait until we add more features like auto-filling for date and time?

File: extensions/ticktick/src/create.tsx (hunk 2 above)
Comment (viduycheung) [reply extensions/ticktick/src/create.tsx]: I get your point. However, my concern is that the majority of users may find it inconvenient to have to go to the bottom of the form to change the project.

**Comment by viduycheung** [comment]
@Alireza29675 I think we can add an action to manually parse, and keep the auto-parsing feature disabled by default.

**Comment by pernielsentikaer** [comment]
@viduycheung Sounds like a plan for a start 🙂

**Review by viduycheung — APPROVED** [review]
Looks good to me 🔥 @Alireza29675 @pernielsentikaer

**Comment by pernielsentikaer** [comment]
Could be cool to have an action to Guess Project, something like:

```
{environment.canAccess(AI) && (
  <Action
    onAction={() => guessProjectByTitle}
    icon={Icon.Wand}
    title="Guess Project"
    shortcut={{ modifiers: ["cmd"], key: "g" }}
  />
)}
```

Do you think we could add that

File: extensions/ticktick/src/create.tsx (hunk 1 above)
Comment (Alireza29675) [reply extensions/ticktick/src/create.tsx]: I guess we can resolve this

File: extensions/ticktick/src/create.tsx (hunk 2 above)
Comment (Alireza29675) [reply extensions/ticktick/src/create.tsx]: I agree that it is wiser decision to keep it as it is 👍 I assume this thread is also resolved

**Comment by Alireza29675** [comment]
@viduycheung @pernielsentikaer I added an action called "Fill the Form with AI" with the hotkey "Command+F". I'm okay with other options; just change the title or hotkey if you feel like it 🙌

**Review by viduycheung — APPROVED** [review]

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**